### PR TITLE
Ffmpeg update 4.2.1

### PR DESCRIPTION
**Pull request by th0ma7**
_Motivation:_  Update of FFMPEG from 4.1 to 4.2 with hope to resolve a few compatibility issues with VideoStation & MediaServer by porting Synology patches.  Also AV1 codec is becoming more popular and can now be enabled.

### PR Description
- [x] Update to latest FFMPEG 4.2.x
- [x] Port Synology default FFMPEG 3.3.7 fixes
- [x] Enable AV1 on supported architectures
- [x] AppArmor vaapi i965_drv_video.so driver fix

### Checklist
- [x] Build rule `all-supported` completed successfully
- [x] Package upgrade completed successfully
- [x] New installation of package completed successfully

_Linked issues:_ #2952, #3413 , #3688, #3716, #3751, #3759

### CODE REVIEW

**Comment by th0ma7** [comment]
Build and package upgrade successful.  Also it does work properly with tvheadend.

Current output:
```
$ /usr/local/ffmpeg/bin/ffmpeg -version 2>/dev/null
ffmpeg version 4.2.1 Copyright (c) 2000-2019 the FFmpeg developers
built with gcc 4.9.3 (crosstool-NG 1.20.0) 20150311 (prerelease)
configuration: --target-os=linux --cross-prefix=/home/th0ma7/git-ffmpeg/spksrc/toolchains/syno-apollolake-6.1/work/x86_64-pc-linux-gnu/bin/x86_64-pc-linux-gnu- --prefix=/var/packages/ffmpeg/target --extra-cflags=-I/home/th0ma7/git-ffmpeg/spksrc/spk/ffmpeg/work-apollolake-6.1/install/var/packages/ffmpeg/target/include --extra-ldflags=-L/home/th0ma7/git-ffmpeg/spksrc/spk/ffmpeg/work-apollolake-6.1/install/var/packages/ffmpeg/target/lib --extra-libs='-lxml2 -ldl' --pkg-config=/usr/bin/pkg-config --ranlib=/home/th0ma7/git-ffmpeg/spksrc/toolchains/syno-apollolake-6.1/work/x86_64-pc-linux-gnu/bin/x86_64-pc-linux-gnu-ranlib --enable-cross-compile --enable-rpath --enable-pic --enable-shared --enable-gpl --enable-fontconfig --enable-libass --enable-libbluray --enable-avresample --enable-libfreetype --enable-libfribidi --enable-libmp3lame --enable-libopus --enable-libsoxr --enable-libvorbis --enable-libvpx --enable-libx264 --enable-libx265 --enable-gnutls --disable-debug --disable-doc --disable-static --arch=x86_64 --enable-thumb --enable-vaapi
libavutil      56. 31.100 / 56. 31.100
libavcodec     58. 54.100 / 58. 54.100
libavformat    58. 29.100 / 58. 29.100
libavdevice    58.  8.100 / 58.  8.100
libavfilter     7. 57.100 /  7. 57.100
libavresample   4.  0.  0 /  4.  0.  0
libswscale      5.  5.100 /  5.  5.100
libswresample   3.  5.100 /  3.  5.100
libpostproc    55.  5.100 / 55.  5.100
```

And available decoders (trucated):
```
$ /usr/local/ffmpeg/bin/ffmpeg -codecs 2>/dev/null | head -15
Codecs:
 D..... = Decoding supported
 .E.... = Encoding supported
 ..V... = Video codec
 ..A... = Audio codec
 ..S... = Subtitle codec
 ...I.. = Intra frame-only codec
 ....L. = Lossy compression
 .....S = Lossless compression
 -------
 D.VI.S 012v                 Uncompressed 4:2:2 10-bit
 D.V.L. 4xm                  4X Movie
 D.VI.S 8bps                 QuickTime 8BPS video
 .EVIL. a64_multi            Multicolor charset for Commodore 64 (encoders: a64multi )
 .EVIL. a64_multi5           Multicolor charset for Commodore 64, extended with 5th color (colram) (encoders: a64multi5 )
...
 DEA.L. ac3                  ATSC A/52A (AC-3) (decoders: ac3 ac3_fixed ) (encoders: ac3 ac3_fixed )
 D.A.L. atrac3               ATRAC3 (Adaptive TRansform Acoustic Coding 3)
 D.A..S atrac3al             ATRAC3 AL (Adaptive TRansform Acoustic Coding 3 Advanced Lossless)
 D.A.L. atrac3p              ATRAC3+ (Adaptive TRansform Acoustic Coding 3+) (decoders: atrac3plus )
 D.A..S atrac3pal            ATRAC3+ AL (Adaptive TRansform Acoustic Coding 3+ Advanced Lossless) (decoders: atrac3plusal )
 DEA.L. eac3                 ATSC A/52B (AC-3, E-AC-3)
...
```

**Comment by m4tt075** [comment]
From past experience with maintaining this package: There are patches in `/cross/ffmpeg/patches`, which have been partly taken from the tvheadend github. Not sure whether they still apply correctly for your updated `ffmpeg` version, but if you haven't done so already, you might want to doublecheck...

**Comment by th0ma7** [comment]
Thnx for the heads-up.
I've looked into the code and none of the patches have been upstreamed.
So ended-up test-applying them manually on latest FFmpeg snapshot to confirm all is still ok, and, surprisingly, they do still apply properly with a few hunk (otherwise `make all-supported` would have failed anyway):
```
th0ma7@debian-dev:~/git-ffmpeg/FFmpeg$ patch -p0 < ../spksrc/cross/ffmpeg/patches/001-replace-arch-env-by-ffmpeg_arch.patch 
patching file configure
Hunk #1 succeeded at 7284 with fuzz 2 (offset 416 lines).
patching file ffbuild/common.mak
Hunk #1 succeeded at 136 with fuzz 2 (offset -1 lines).
patching file Makefile
Hunk #1 succeeded at 86 with fuzz 2 (offset 3 lines).
th0ma7@debian-dev:~/git-ffmpeg/FFmpeg$ patch -p0 < ../spksrc/cross/ffmpeg/patches/002-tvheadend-ffmpeg.libx264.patch 
patching file ./libavcodec/libx264.c
Hunk #1 succeeded at 1041 (offset 139 lines).
th0ma7@debian-dev:~/git-ffmpeg/FFmpeg$ patch -p0 < ../spksrc/cross/ffmpeg/patches/003-tvheadend-ffmpeg.libx265.patch 
patching file ./libavcodec/libx265.c
Hunk #1 succeeded at 387 (offset 123 lines).
Hunk #2 succeeded at 443 with fuzz 2 (offset 130 lines).
th0ma7@debian-dev:~/git-ffmpeg/FFmpeg$ patch -p0 < ../spksrc/cross/ffmpeg/patches/004-tvheadend-ffmpeg.vaapi_encode.patch 
patching file ./libavcodec/vaapi_encode.c
Hunk #1 succeeded at 26 with fuzz 2.
Hunk #2 succeeded at 580 (offset 125 lines).
Hunk #3 succeeded at 612 (offset 125 lines).
```

**Comment by ymartin59** [comment]
Hello @th0ma7 Yet another challenge for you.
Synology has patched ffmpeg to add command line option and "jump" feature. But "forgot" to publish patches for version in latest DSM 6.2 VideoStation: https://sourceforge.net/projects/dsgpl/files/Synology%20NAS%20GPL%20Source/22259branch/x64-source/
If you may contact them to request for GPL sources update publication, it would help to report patch on 4.2.x
Idea is to solve #3413 - which also requires as prerequisite to add proper apparmor configuration in  VideoStation to allow SynoCommunity ffmpeg to be loaded by process.
Regards. Yves

**Comment by th0ma7** [comment]
I can certainly reach out to Synology but trying to figure out what the issue is to know what I am asking for....  Also is it a blocker to get this version out?

Default syno `ffmpeg` show the following:
```
$ ffmpeg -codecs 2>/dev/null
Codecs:
 D..... = Decoding supported
 .E.... = Encoding supported
 ..V... = Video codec
 ..A... = Audio codec
 ..S... = Subtitle codec
 ...I.. = Intra frame-only codec
 ....L. = Lossy compression
 .....S = Lossless compression
 -------
 ..A.LS dts                  DCA (DTS Coherent Acoustics)
```
And using latest build version 4.2.1:
```
$ /usr/local/ffmpeg/bin/ffmpeg -codecs 2>/dev/null
Codecs:
 D..... = Decoding supported
 .E.... = Encoding supported
 ..V... = Video codec
 ..A... = Audio codec
 ..S... = Subtitle codec
 ...I.. = Intra frame-only codec
 ....L. = Lossy compression
 .....S = Lossless compression
 -------
 D.A.L. dolby_e              Dolby E
 DEA.LS dts                  DCA (DTS Coherent Acoustics) (decoders: dca ) (encoders: dca )
```

**Comment by th0ma7** [comment]
Ok, got answers from Synology.
Source code is located here:
https://sourceforge.net/projects/dsgpl/files/Packages/DSM%206.1%20Package%20Release/

Both MediaServer and VideoStation has a copy of the Synology patched ffmpeg version 3.3.7 with only minor differences between them that can easily be merged together.

The fun part: quick comparison of the code with the original source tree (diff -uprNB --exclude=.gitignore) show the following:
- Addition of a `configure.syno`
- Addition of `#include "synoconfig.h"`
- Interesting addition of a `#include "synog1codecpatent.h"`
- Addition of various bits of code on audio skip thingy on a few `.c`

Most interesting, output patch doesn't apply as is over 4.2.1 but there isn't that many rejects... Worth investigating I guess.  I'll see what i can do to create a patch file over 4.2.1.

**Comment by th0ma7** [comment]
I was able to create a first patch without too much effort.
There was one change I didn't port although kept the related `.rej` for future reference as needed.
[synology-ffmpeg-4.2.1-media-and-video.patch.gz](https://github.com/SynoCommunity/spksrc/files/3731973/synology-ffmpeg-4.2.1-media-and-video.patch.gz)

Building is in progress, wait & see.

**Comment by th0ma7** [comment]
Compiling & package building looking good.
New patches updated in my tree available for pull.
Needs testing from people able to trigger that specific bug to confirm it resolves the issue.
As needed I probably could provided packages at a temporary location for people to test things out?

**Comment by th0ma7** [comment]
Temporary testing packages available here:
https://github.com/th0ma7/synology/tree/master/packages

**Notes:** Contains FFMPEG v4.2.1 + synology patch port but does not contain AV1 codec support yet.

**Comment by th0ma7** [comment]
Having AV1 now working on supported arch.
Testing packages to be updated once build is complete for all platforms.
```
$ /usr/local/ffmpeg/bin/ffmpeg -codecs 2>/dev/null
Codecs:
 D..... = Decoding supported
 .E.... = Encoding supported
 ..V... = Video codec
 ..A... = Audio codec
 ..S... = Subtitle codec
 ...I.. = Intra frame-only codec
 ....L. = Lossy compression
 .....S = Lossless compression
 -------
 DEV.L. av1                  Alliance for Open Media AV1 (decoders: libaom-av1 ) (encoders: libaom-av1 )
```
And the AV1 `aom` related encoder/decoder:
```
$ ls -la /usr/local/ffmpeg/bin/aom*
-rwxr-xr-x 1 root root 610576 Oct 18 09:07 /usr/local/ffmpeg/bin/aomdec
-rwxr-xr-x 1 root root 663808 Oct 18 09:07 /usr/local/ffmpeg/bin/aomenc
```

File: cross/libaom/Makefile [hunk 1]
```diff
@@ -0,0 +1,96 @@
...
+
+.PHONY: libaom_custom_compile
+
+libaom_custom_compile: 
+	cd $(PKG_WORK_DIR) && env $(ENV) $(MAKE)
+
+.PHONY: libaom_custom_install
+
+libaom_custom_install: 
+	cd $(PKG_WORK_DIR) && env $(ENV) $(MAKE) install DESTDIR=$(INSTALL_DIR)
```
Comment (ymartin59) [inline cross/libaom/Makefile]: That is the first time CMake sources are built that way in a dedicated build path.
Until now, sources are built "in place" in current source directory as a dedicated one is created for each architecture.
So the question: is it really worth to build "elsewhere" as our source directory cannot be considered as "developer workspace" but only temporary discarded compilation tree...

File: cross/libaom/PLIST [hunk 2]
```diff
@@ -0,0 +1,15 @@
...
+rsc:include/aom/aom.h
+rsc:include/aom/aom_codec.h
+rsc:include/aom/aom_frame_buffer.h
+rsc:include/aom/aom_image.h
+rsc:include/aom/aom_integer.h
+rsc:include/aom/aom.h
+rsc:include/aom/aom_decoder.h
+rsc:include/aom/aomdx.h
+rsc:include/aom/aomcx.h
+rsc:include/aom/aom_encoder.h
```
Comment (ymartin59) [inline cross/libaom/PLIST]: There is no need to include headers and pkgconfig files in generated package

File: spk/ffmpeg/Makefile [hunk 3]
```diff
@@ -1,8 +1,8 @@
 SPK_NAME = ffmpeg
-SPK_VERS = 4.1
-SPK_REV = 11
+SPK_VERS = 4.2.1
+SPK_REV = 13
 SPK_ICON = src/ffmpeg.png
-CHANGELOG = "1. Update ffmpeg to 4.1<br/>2. Update ffmpeg dependencies<br/>3. Add VAAPI for intel x64 processors<br/>4. Fix HDR10Plus support in x265<br/>5. Fix x265 support for ppc platforms"
+CHANGELOG = "1. Add AOMedia Video 1 (AV1) for ARVv7, ARMv8 and x86 platforms<br/>2. Update ffmpeg to 4.2.1<br/>3. Port Synology FFMPEG 3.3.7 changes and fixes for VideoStation and MediaServer<br/>4. Update ffmpeg dependencies<br/>5. Add VAAPI for intel x64 processors<br/>6. Fix HDR10Plus support in x265<br/>5. Fix x265 support for ppc platforms"
```
Comment (ymartin59) [inline spk/ffmpeg/Makefile]:
```suggestion
CHANGELOG = "1. Add AOMedia Video 1 (AV1) for ARMv7, ARMv8 and x86 platforms<br/>2. Update ffmpeg to 4.2.1<br/>3. Port Synology FFMPEG 3.3.7 changes and fixes for VideoStation and MediaServer<br/>"
```

ChangeLog should only refer to this specific package revision

File: cross/libaom/Makefile (hunk 1 above)
Comment (th0ma7) [reply cross/libaom/Makefile]: First time for me as well.  I had to play with it quite a bit to have it working as libaom doesn't support in-directory building:
```
CMake Error at CMakeLists.txt:27 (message):
  Building from within the aom source tree is not supported.
```
Which explains why I ended-up building "elsewhere" ... Alternative ideas on this welcomed.

File: cross/libaom/PLIST (hunk 2 above)
Comment (th0ma7) [reply cross/libaom/PLIST]: Indeed.  I had found an issue in my earlier builds where pkgconfig file was creating errors but I guess that was due to the version mismatch at the time.  I tried removing the headers & pkgconfig files from the `PLIST` and re-did a test build on selected arch successfully.  Changes now committed.

**Comment by sotux** [comment]
Hi @th0ma7 , I would test your ffmpeg package. But there seems no braswell package for my J3160. Which one should I download?
Thanks.

**Comment by th0ma7** [comment]
`braswell` platform fits under the `x86_64` arch.  Give that one a try.

**Comment by SLboat** [comment]
thank you great work,but how this work in dsm 6.22? i use one patch change ffmpeg to this version ,but no luck with seck time..

**Comment by th0ma7** [comment]
Hopefuly I'm partially answering to your questions:
> thank you great work,but how this work in dsm 6.22?

* Is it compatible with DSM 6.2.2: yes

> i use one patch change ffmpeg to this version ,but no luck with seck time..
* patch?  I'm sorry but I have no idea what you are referring to?  git patches?  pre-built packages?

**Comment by SLboat** [comment]
Hi,I’m use the ds918+,Dan 6.22.and I installed the ffmpeg 4.2package from your test repo, but it still can’t decode the dots audio ,so is it something I not doing right?

Sorry for my poor English,


From SLboat

**Comment by SLboat** [comment]
I’m use j1900cpu, but installed ds918 system ,so should I use the x86 version instead apollolake version?


Here is what I have done:
1. Install the newest video station on dsm
2. Install the new ffmpeg from 

https://github.com/th0ma7/synology/tree/master/packages

3. In video station open a dots video ,it still say can’t decode
Just like before 


From SLboat

**Comment by th0ma7** [comment]
DS918+ version has an apollolake processor.
Therefore you can either use the x64 package or the arch specific apollolake package.
As for dots video, I guess you are referring to DTS ?  If so I suggest you look at bug entry #2952 where some people might be able to help you out on this issue specifically.

To me question is, are you able to decode it on the command line directly?
Because the problem may be with ffmpeg or it's integration with videostation.

**Comment by SLboat** [comment]
I’m see now,I had see the issue,and use the script success play dts from beginning,but when I drag the time progress bar,it’s lack and cpu 100% 


From SLboat

**Comment by SLboat** [comment]
> DS918+ version has an apollolake processor.
> Therefore you can either use the x64 package or the arch specific apollolake package.
> As for dots video, I guess you are referring to DTS ? If so I suggest you look at bug entry #2952 where some people might be able to help you out on this issue specifically.
> 
> To me question is, are you able to decode it on the command line directly?
> Because the problem may be with ffmpeg or it's integration with videostation.

i think i know why we need this project now...the community ffmpeg is not as a fix for video staion, it's a package, fit for ffmpeg.

**Comment by SLboat** [comment]
> DS918+ version has an apollolake processor.
> Therefore you can either use the x64 package or the arch specific apollolake package.
> As for dots video, I guess you are referring to DTS ? If so I suggest you look at bug entry #2952 where some people might be able to help you out on this issue specifically.
> 
> To me question is, are you able to decode it on the command line directly?
> Because the problem may be with ffmpeg or it's integration with videostation.

sorry, i don't know how to use the command line to test ffmpeg, i think the dts decoder is work well, the problem is the video station.

```bash
$ ffmpeg -codecs 2>/dev/null | grep -E 'ac3|dts'
 
 ..A.L. ac3                  ATSC A/52A (AC-3)
 D.A.L. atrac3               ATRAC3 (Adaptive TRansform Acoustic Coding 3)
 D.A.L. atrac3p              ATRAC3+ (Adaptive TRansform Acoustic Coding 3+) (decoders: atrac3plus )
 ..A.LS dts                  DCA (DTS Coherent Acoustics)
 ..A.L. eac3                 ATSC A/52B (AC-3, E-AC-3)
```

```bash
# /var/packages/VideoStation/target/bin/ffmpeg -codecs 2>/dev/null | grep -E 'ac3|dts'

 D.A.L. ac3                  ATSC A/52A (AC-3) (decoders: ac3 ac3_fixed )
 D.A.L. atrac3               ATRAC3 (Adaptive TRansform Acoustic Coding 3)
 D.A..S atrac3al             ATRAC3 AL (Adaptive TRansform Acoustic Coding 3 Advanced Lossless)
 D.A.L. atrac3p              ATRAC3+ (Adaptive TRansform Acoustic Coding 3+) (decoders: atrac3plus )
 D.A..S atrac3pal            ATRAC3+ AL (Adaptive TRansform Acoustic Coding 3+ Advanced Lossless) (decoders: atrac3plusal )
 ..A.LS dts                  DCA (DTS Coherent Acoustics)
 ..A.L. eac3                 ATSC A/52B (AC-3, E-AC-3)
```

```bash
$ /usr/local/ffmpeg/bin/ffmpeg -codecs 2>/dev/null | grep -E 'ac3|dts'

 DEA.L. ac3                  ATSC A/52A (AC-3) (decoders: ac3 ac3_fixed ) (encoders: ac3 ac3_fixed )
 D.A.L. atrac3               ATRAC3 (Adaptive TRansform Acoustic Coding 3)
 D.A..S atrac3al             ATRAC3 AL (Adaptive TRansform Acoustic Coding 3 Advanced Lossless)
 D.A.L. atrac3p              ATRAC3+ (Adaptive TRansform Acoustic Coding 3+) (decoders: atrac3plus )
 D.A..S atrac3pal            ATRAC3+ AL (Adaptive TRansform Acoustic Coding 3+ Advanced Lossless) (decoders: atrac3plusal )
 DEA.LS dts                  DCA (DTS Coherent Acoustics) (decoders: dca ) (encoders: dca )
 DEA.L. eac3                 ATSC A/52B (AC-3, E-AC-3)
 ```

**Comment by th0ma7** [comment]
@SLboat please open-up a different issue for that problem specifically.
And again, have a look at issue #2952 which may be related somehow.

**Comment by SLboat** [comment]
I see, Thank you !


From SLboat

**Comment by th0ma7** [comment]
AppArmor fix added.
@ymartin59 anything else missing for a relase?

File: cross/libaom/Makefile (hunk 1 above)
Comment (ymartin59) [reply cross/libaom/Makefile]: Noticed. I will create a support for that in framework... one day...

File: spk/ffmpeg/Makefile [hunk 4]
```diff
@@ -1,13 +1,14 @@
 SPK_NAME = ffmpeg
-SPK_VERS = 4.1
-SPK_REV = 11
+SPK_VERS = 4.2.1
+SPK_REV = 14
 SPK_ICON = src/ffmpeg.png
-CHANGELOG = "1. Update ffmpeg to 4.1<br/>2. Update ffmpeg dependencies<br/>3. Add VAAPI for intel x64 processors<br/>4. Fix HDR10Plus support in x265<br/>5. Fix x265 support for ppc platforms"
+CHANGELOG = "1. apparmor permission denied fix<br/>2. Add AOMedia Video 1 (AV1) for ARMv7, ARMv8 and x86 platforms<br/>3. Update ffmpeg to 4.2.1<br/>4. Port Synology FFMPEG 3.3.7 changes and fixes for VideoStation and MediaServer"
```
Comment (ymartin59) [inline spk/ffmpeg/Makefile]:
```suggestion
CHANGELOG = "1. Update ffmpeg to 4.2.1<br/>2. Add AOMedia Video 1 (AV1) for ARMv7, ARMv8 and x86 platforms<br/>3. apparmor permission denied fix<br/>4. Port Synology FFMPEG 3.3.7 changes and fixes for VideoStation and MediaServer"
```

**Comment by ymartin59** [comment]
@th0ma7 In that situation, I prefer "squash and merge" to get a "cleaner" git history/log and normalize commit... Now it is done, do not try anything to fix.

**Comment by kref** [comment]
this build failed to decode some dts streams, following is a working build, any idea?
```
ffmpeg version 4.2.1-static https://johnvansickle.com/ffmpeg/  Copyright (c) 2000-2019 the FFmpeg developers
  built with gcc 6.3.0 (Debian 6.3.0-18+deb9u1) 20170516
  configuration: --enable-gpl --enable-version3 --enable-static --disable-debug --disable-ffplay --disable-indev=sndio --disable-outdev=sndio --cc=gcc-6 --enable-fontconfig --enable-frei0r --enable-gnutls --enable-gmp --enable-libgme --enable-gray --enable-libaom --enable-libfribidi --enable-libass --enable-libvmaf --enable-libfreetype --enable-libmp3lame --enable-libopencore-amrnb --enable-libopencore-amrwb --enable-libopenjpeg --enable-librubberband --enable-libsoxr --enable-libspeex --enable-libsrt --enable-libvorbis --enable-libopus --enable-libtheora --enable-libvidstab --enable-libvo-amrwbenc --enable-libvpx --enable-libwebp --enable-libx264 --enable-libx265 --enable-libxml2 --enable-libdav1d --enable-libxvid --enable-libzvbi --enable-libzimg
  libavutil      56. 31.100 / 56. 31.100
  libavcodec     58. 54.100 / 58. 54.100
  libavformat    58. 29.100 / 58. 29.100
  libavdevice    58.  8.100 / 58.  8.100
  libavfilter     7. 57.100 /  7. 57.100
  libswscale      5.  5.100 /  5.  5.100
  libswresample   3.  5.100 /  3.  5.100
  libpostproc    55.  5.100 / 55.  5.100
```

**Comment by th0ma7** [comment]
> this build failed to decode some dts streams, following is a working build, any idea?
> 
> ```
> ffmpeg version 4.2.1-static https://johnvansickle.com/ffmpeg/  Copyright (c) 2000-2019 the FFmpeg developers
>   built with gcc 6.3.0 (Debian 6.3.0-18+deb9u1) 20170516
>   configuration: --enable-gpl --enable-version3 --enable-static --disable-debug --disable-ffplay --disable-indev=sndio --disable-outdev=sndio --cc=gcc-6 --enable-fontconfig --enable-frei0r --enable-gnutls --enable-gmp --enable-libgme --enable-gray --enable-libaom --enable-libfribidi --enable-libass --enable-libvmaf --enable-libfreetype --enable-libmp3lame --enable-libopencore-amrnb --enable-libopencore-amrwb --enable-libopenjpeg --enable-librubberband --enable-libsoxr --enable-libspeex --enable-libsrt --enable-libvorbis --enable-libopus --enable-libtheora --enable-libvidstab --enable-libvo-amrwbenc --enable-libvpx --enable-libwebp --enable-libx264 --enable-libx265 --enable-libxml2 --enable-libdav1d --enable-libxvid --enable-libzvbi --enable-libzimg
>   libavutil      56. 31.100 / 56. 31.100
>   libavcodec     58. 54.100 / 58. 54.100
>   libavformat    58. 29.100 / 58. 29.100
>   libavdevice    58.  8.100 / 58.  8.100
>   libavfilter     7. 57.100 /  7. 57.100
>   libswscale      5.  5.100 /  5.  5.100
>   libswresample   3.  5.100 /  3.  5.100
>   libpostproc    55.  5.100 / 55.  5.100
> ```

Hi @kref , there are so many differences in the build you are referring to that there is sadly nowhere to start from.  Please submit a new issue if you think there is something worth looking into and don't forget to provide details of your platform, DSM version, ffmpeg version and details, how you used it (natively or through "modified" videostation) along with a sample of the file that refuses to play.

Thnx and goodluck!

**Comment by kref** [comment]
> > this build failed to decode some dts streams, following is a working build, any idea?
> > ```
> > ffmpeg version 4.2.1-static https://johnvansickle.com/ffmpeg/  Copyright (c) 2000-2019 the FFmpeg developers
> >   built with gcc 6.3.0 (Debian 6.3.0-18+deb9u1) 20170516
> >   configuration: --enable-gpl --enable-version3 --enable-static --disable-debug --disable-ffplay --disable-indev=sndio --disable-outdev=sndio --cc=gcc-6 --enable-fontconfig --enable-frei0r --enable-gnutls --enable-gmp --enable-libgme --enable-gray --enable-libaom --enable-libfribidi --enable-libass --enable-libvmaf --enable-libfreetype --enable-libmp3lame --enable-libopencore-amrnb --enable-libopencore-amrwb --enable-libopenjpeg --enable-librubberband --enable-libsoxr --enable-libspeex --enable-libsrt --enable-libvorbis --enable-libopus --enable-libtheora --enable-libvidstab --enable-libvo-amrwbenc --enable-libvpx --enable-libwebp --enable-libx264 --enable-libx265 --enable-libxml2 --enable-libdav1d --enable-libxvid --enable-libzvbi --enable-libzimg
> >   libavutil      56. 31.100 / 56. 31.100
> >   libavcodec     58. 54.100 / 58. 54.100
> >   libavformat    58. 29.100 / 58. 29.100
> >   libavdevice    58.  8.100 / 58.  8.100
> >   libavfilter     7. 57.100 /  7. 57.100
> >   libswscale      5.  5.100 /  5.  5.100
> >   libswresample   3.  5.100 /  3.  5.100
> >   libpostproc    55.  5.100 / 55.  5.100
> > ```
> 
> Hi @kref , there are so many differences in the build you are referring to that there is sadly nowhere to start from. Please submit a new issue if you think there is something worth looking into and don't forget to provide details of your platform, DSM version, ffmpeg version and details, how you used it (natively or through "modified" videostation) along with a sample of the file that refuses to play.
> 
> Thnx and goodluck!

Hi @th0ma7 , further investigation suggests it is a GCC bug, rather than a configuration issue I suspected.

**Pull request merged**